src/pages/ReportSettingsPage.js

```diff
@@ -0,0 +1,236 @@
+import _ from 'underscore';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+import styles from '../styles/styles';
+import compose from '../libs/compose';
+import Navigation from '../libs/Navigation/Navigation';
+import * as Report from '../libs/actions/Report';
+import * as ReportUtils from '../libs/reportUtils';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import ScreenWrapper from '../components/ScreenWrapper';
+import withLocalize, {withLocalizePropTypes} from '../components/withLocalize';
+import Text from '../components/Text';
+import Button from '../components/Button';
+import TextInputWithPrefix from '../components/TextInputWithPrefix';
+import Picker from '../components/Picker';
+import withFullPolicy, {fullPolicyDefaultProps, fullPolicyPropTypes} from './workspace/withFullPolicy';
+
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** The active report */
+    report: PropTypes.shape({
+        /** The list of icons */
+        icons: PropTypes.arrayOf(PropTypes.string),
+
+        /** The report name */
+        reportName: PropTypes.string,
+
+        /** ID of the report */
+        reportID: PropTypes.number,
+
+        /** The current user's notification preference for this report */
+        notificationPreference: PropTypes.string,
+    }).isRequired,
+
+    /** Route params */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** Report ID passed via route r/:reportID/settings */
+            reportID: PropTypes.string,
+        }),
+    }).isRequired,
+
+    ...fullPolicyPropTypes,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    ...fullPolicyDefaultProps,
+};
+
+
+class ReportSettingsPage extends Component {
+    constructor(props) {
+        super(props);
+        this.notificationPreferencesOptions = {
+            default: {
+                value: CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS,
+                label: this.props.translate('notificationPreferences.always'),
+            },
+            daily: {
+                value: CONST.REPORT.NOTIFICATION_PREFERENCE.DAILY,
+                label: this.props.translate('notificationPreferences.daily'),
+            },
+            mute: {
+                value: CONST.REPORT.NOTIFICATION_PREFERENCE.MUTE,
+                label: this.props.translate('notificationPreferences.mute'),
+            },
+        };
+
+        this.state = {
+            newRoomName: this.props.report.reportName,
+        };
+    }
+
+    /**
+     * Modifies the room name to follow our conventions:
+     * - Max length 80 characters
+     * - Cannot not include space or special characters, and we automatically apply an underscore for spaces
+     * - Must be lowercase
+     * Also checks to see if this room name already exists, and displays an error message if so.
+     * @param {String} roomName
+     *
+     * @returns {String}
+     */
+    checkAndModifyRoomName(roomName) {
```

Comment:
Looks like we're just copying this from `src/pages/workspace/WorkspaceNewRoomPage.js` - should we put this in a common place and import it into both files instead?

Reply:
This is a good point. It would be cleaner this way. Let me see if I can figure it out real quick.